packages/waas/src/auth.ts

```diff
@@ -339,9 +339,21 @@ export class SequenceWaaS {
     }
   }
 
-  private async updateTimeDrift() {
+  /**
+   * Checks the server status and sets the time drift before sending any intent.
+   * @throws {Error} If server status check fails or Date header is missing
+   */
+  private async preSendIntent() {
     if (getTimeDrift() === undefined) {
       const res = await fetch(`${this.config.rpcServer}/status`)
+
+      if (res.status !== 200) {
+        if (res.status === 451) {
+          throw new Error('Unavailable For Legal Reasons')
```

Comment:
Can you update this error to say: "Service unavailable due to legal and geographic restrictions"